src/mpi/coll/algorithms/common/coll_schedule_tree.h

```diff
@@ -164,9 +164,10 @@ COLL_sched_bcast_tree_pipelined(void *buffer, int count, COLL_dt_t datatype, int
     for (i = 0; i < num_chunks; i++) {
         int msgsize = (i < num_chunks_floor) ? chunk_size_floor : chunk_size_ceil;
         mpi_errno =
-            COLL_sched_bcast_tree((char *) buffer + offset * extent, msgsize, datatype, root, tag,
+            COLL_sched_bcast_tree((char *) buffer + offset * extent, msgsize, datatype, root, *tag,
                                   comm, tree_type, k, sched, 0);
         offset += msgsize;
+        (*tag)++;
```

Comment:
I'm skeptical of this patch overall. We should have some number of ordering tests running regularly with `ch4/ucx` and haven't seen any issues. A small reproducer would go a long way in convincing me...